pandas/indexes/multi.py

```diff
@@ -1007,6 +1008,9 @@ def from_tuples(cls, tuples, sortorder=None, names=None):
             # I think this is right? Not quite sure...
             raise TypeError('Cannot infer number of levels from empty list')
 
+        if not _check_equal_length(tuples):
```

Comment:
I would definitely consider doing this check in Cython at the same time as constructing the arrays in the `lib.tuples_to_object_array` and `lib.to_object_array_tuples` functions. Otherwise I would expect a serious impact on performance in the typical case.